### PR TITLE
Clean up Javadoc, and add a Javadoc build target

### PR DESCRIPTION
**Pull request by hegemonic**
This pull request includes some minor Javadoc cleanup, including:

1. For completeness, document things like getters/setters/builders.
2. Remove a couple of implementation details (for example, the fact that we use protobufs in the coders).
3. Flag a couple of methods that shouldn't be called directly. (There are probably more methods in this category; I tried to get the ones that seemed most likely to confuse people.)

I also set up a Javadoc build target, including automatic links to the Dataflow and HBase API docs.

### CODE REVIEW

**Comment by sduskis** [comment]
Please do the following to fix the merge conflicts:  

git remote add upstream https://github.com/GoogleCloudPlatform/cloud-bigtable-client.git
git fetch upstream
git rebase upstream/master

**Comment by hegemonic** [comment]
Fixed!

File: bigtable-hbase-dataflow/pom.xml [hunk 1]
```diff
@@ -138,6 +138,36 @@ limitations under the License.
                     </execution>
                 </executions>
             </plugin>
+
```
Comment (sduskis) [inline bigtable-hbase-dataflow/pom.xml]: This plugin should be wrapped in the "release" profile rather than run for every build.  I'll do that post merge.

**Comment by sduskis** [comment]
LGTM.  

I'll take care of moving the javadoc generation to the "release" profile.

**Pull request merged**